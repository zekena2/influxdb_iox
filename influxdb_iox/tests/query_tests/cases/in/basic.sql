-- Basic query tests
-- IOX_SETUP: TwoMeasurements

-- query data
SELECT * from cpu;

SELECT min(region) from cpu;

-- projection
-- expect that to get a subset of the columns and in the order specified
SELECT "user", region from cpu;

-- predicate on CPU
SELECT * from cpu where time > to_timestamp('1970-01-01T00:00:00.000000120+00:00');

-- predicate on CPU with explicit coercion (cast string to timestamp)
SELECT * from cpu where time > '1970-01-01T00:00:00'::timestamp ORDER BY time;

-- predicate on CPU with automatic coercion (comparing time to string)
SELECT * from cpu where time > '1970-01-01T00:00:00' ORDER BY time;

-- projection and predicate
-- expect that to get a subset of the columns and in the order specified
SELECT "user", region from cpu where time > to_timestamp('1970-01-01T00:00:00.000000120+00:00');

-- Functions on tags
--
SELECT upper("region"), substring(region, 1, 2) from cpu where time > to_timestamp('1970-01-01T00:00:00.000000120+00:00');

-- basic grouping
SELECT count(*) from cpu group by region;

-- select from a different measurement
SELECT * from disk;

-- MEDIAN should work
select MEDIAN("user"), region from cpu group by region;
