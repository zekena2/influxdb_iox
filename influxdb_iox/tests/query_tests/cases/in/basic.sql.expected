-- Test Setup: TwoMeasurements
-- SQL: SELECT * from cpu;
+--------+--------------------------------+------+
| region | time                           | user |
+--------+--------------------------------+------+
| west   | 1970-01-01T00:00:00.000000100Z | 23.2 |
| west   | 1970-01-01T00:00:00.000000150Z | 21.0 |
+--------+--------------------------------+------+
-- SQL: SELECT min(region) from cpu;
+-----------------+
| MIN(cpu.region) |
+-----------------+
| west            |
+-----------------+
-- SQL: SELECT "user", region from cpu;
+------+--------+
| user | region |
+------+--------+
| 23.2 | west   |
| 21.0 | west   |
+------+--------+
-- SQL: SELECT * from cpu where time > to_timestamp('1970-01-01T00:00:00.000000120+00:00');
+--------+--------------------------------+------+
| region | time                           | user |
+--------+--------------------------------+------+
| west   | 1970-01-01T00:00:00.000000150Z | 21.0 |
+--------+--------------------------------+------+
-- SQL: SELECT * from cpu where time > '1970-01-01T00:00:00'::timestamp ORDER BY time;
+--------+--------------------------------+------+
| region | time                           | user |
+--------+--------------------------------+------+
| west   | 1970-01-01T00:00:00.000000100Z | 23.2 |
| west   | 1970-01-01T00:00:00.000000150Z | 21.0 |
+--------+--------------------------------+------+
-- SQL: SELECT * from cpu where time > '1970-01-01T00:00:00' ORDER BY time;
+--------+--------------------------------+------+
| region | time                           | user |
+--------+--------------------------------+------+
| west   | 1970-01-01T00:00:00.000000100Z | 23.2 |
| west   | 1970-01-01T00:00:00.000000150Z | 21.0 |
+--------+--------------------------------+------+
-- SQL: SELECT "user", region from cpu where time > to_timestamp('1970-01-01T00:00:00.000000120+00:00');
+------+--------+
| user | region |
+------+--------+
| 21.0 | west   |
+------+--------+
-- SQL: SELECT upper("region"), substring(region, 1, 2) from cpu where time > to_timestamp('1970-01-01T00:00:00.000000120+00:00');
+-------------------+--------------------------------------+
| upper(cpu.region) | substr(cpu.region,Int64(1),Int64(2)) |
+-------------------+--------------------------------------+
| WEST              | we                                   |
+-------------------+--------------------------------------+
-- SQL: SELECT count(*) from cpu group by region;
+----------+
| COUNT(*) |
+----------+
| 2        |
+----------+
-- SQL: SELECT * from disk;
+-------+--------+--------------------------------+
| bytes | region | time                           |
+-------+--------+--------------------------------+
| 99    | east   | 1970-01-01T00:00:00.000000200Z |
+-------+--------+--------------------------------+
-- SQL: select MEDIAN("user"), region from cpu group by region;
+------------------+--------+
| MEDIAN(cpu.user) | region |
+------------------+--------+
| 22.1             | west   |
+------------------+--------+